test/test_helper.rb

```diff
@@ -83,6 +84,8 @@ def teardown
 include TestHelpers::XmlAssertions
 include TestHelpers::SectionsPermissions
 
+include ActiveSupport::Testing::TimeHelpers
```

Comment:
I think we need to `prepend` this because it contains an `after_teardown` method. Then line 74 `travel_back` can be removed.

Reply:
```suggestion
prepend ActiveSupport::Testing::TimeHelpers
```

When prepending, then the module methods will override methods define in the including module. What we want afaict. If it is included, it will not override such methods.

Reply:
I understand. But in `TimeHelpers` the method is `after_teardown`, while ours is `teardown`. They shouldn't override each other in any order, and as `travel_back` in `after_teardown` is being called anyway, there's no need to call `travel_back` in our `teardown` method. Or am I missing something?

The fact is I already pushed changes that remove the call to `travel_back` from `teardown`, [705b75](https://github.com/3scale/porta/pull/3062/commits/705b75aa71c7f086411db825ec88528ddd1d362b) and all [tests](https://app.circleci.com/pipelines/github/3scale/porta/21608/workflows/d19940b9-dd30-4b2d-ae2b-bf3efd09e2b7) passed.

Reply:
wait, this includes it in `Object` or something. Why not prepend it in `ActiveSupport::TestCase`?

and while on it, how do you make sure that `#after_teardown` was actually called?

Reply:
That includes the [TimeHelpers](https://github.com/rails/rails/blob/8015c2c2cf5c8718449677570f372ceb01318a32/activesupport/lib/active_support/testing/time_helpers.rb#L69) module into `test_helper.rb`, so it's available for all Minitest tests. Same for Rspec and `rails_helper.rb`. I set a breakpoint in `after_teardown` and ran unit tests and specs, it stopped in the breakpoint after each test.

Reply:
It apparently works somehow but I'm not convinced it works in the way it is supposed to. I prefer to understand how it works and remove redundant `#teardown` methods in individual classes.

If I open a pry shell in a test I see:

```
[14] pry(#<UserTest::ExistingProviderUserTest>)> ls self
...
ActiveSupport::Testing::TimeHelpers#methods: 
  freeze_time  travel  travel_back  travel_to
...
ActiveJob::TestHelper#methods: 
  after_teardown            assert_performed_jobs  perform_enqueued_jobs 
  assert_enqueued_jobs      assert_performed_with  performed_jobs        
  assert_enqueued_with      before_setup           performed_jobs=       
  assert_no_enqueued_jobs   enqueued_jobs          queue_adapter         
  assert_no_performed_jobs  enqueued_jobs=         queue_adapter_for_test
...
```

I read this that `#after_teardown` comes from `ActiveJob::TestHelper` instead of `ActiveSupport::Testing::TimeHelpers`.

In fact `TimeHelpers` is somehow mixed in regardless of `include ActiveSupport::Testing::TimeHelpers` inside `test/test_helper.rb`. So the `include` lines should be inside class `ActiveSupport::TestCase` in general. Just in this case it seems included in another way.

Reply:
I removed all redundant callbacks to restore the time. Some notes:

- Everything inheriting from `ActiveSupport::TestCase` gets the teardown called automatically. So no need to write any callback manually as well as requiring/including it at `test_helper.rb`.
- `ActionDispatch::IntegrationTest` inherits from `ActiveSupport::TestCase`
- I modified `SimpleMiniTest` to inherit from `ActiveSupport::TestCase`. It was inheriting from `Minitest::Test` and the teardown was not being called automatically. Since `ActiveSupport::TestCase < Minitest::Test` it shouldn't be any problem.
- In specs, tests doesn't inherit from `ActiveSupport::TestCase`. So including `ActiveSupport::Testing::TimeHelpers` in `rails_helper.rb` is needed to get the teardown called for specs.
- Cucmbers doesn't inherit from  `ActiveSupport::TestCase` neither. So a manual teardown is needed at [features/step_definitions/timehelpers_steps.rb#L4](https://github.com/3scale/porta/blob/90afeec0ea33df5a1decbc372e19169aea26af6b/features/step_definitions/timehelpers_steps.rb#L4)
- I removed the setup method to freeze time from [test/unit/concerns/symbolize_test.rb](https://github.com/3scale/porta/pull/3062/files#diff-decc2dab97a7fc9aea3174f2f1bd6eab30f283349f6a13f6c362c298006c5c2aL20). It doesn't seem to me that freezing time there is required. And these tests pass locally after removing the callback.
- `after_teardown` at `ActiveJob::TestHelper` ends up calling `after_teardown` at `ActiveSupport::Testing::TimeHelpers`. See the attached image.

![image](https://user-images.githubusercontent.com/17052241/194855472-88c68880-f492-47c3-b62a-468ccd3525a6.png)